grails-app/services/org/pih/warehouse/data/ProductSupplierDataService.groovy

```diff
@@ -302,7 +302,8 @@ class ProductSupplierDataService {
                 'manufacturerCode': manufacturerCode,
                 'supplierCode': supplierCode,
         ])
-        def productSupplier = data ? ProductSupplier.get(data.first().id) : null
+        ArrayList<ProductSupplier> productSuppliers = data?.collect{ it -> ProductSupplier.get(it?.id) }?.sort{ !it?.active }
```

Comment:
Could you add a comment to describe what you're trying to accomplish here? Do we really want to sort by active? And in this case, what does that do? Sorts active = true first, then action = false? What about active = null?

Reply:
I'll add a comment - it just "pushes" an active one to the top if there are > 1 product sources where one or more of them could be inactive - I agree that anyway it can return inactive one if there are for example two inactive in the list, but the case here is to return active if there is for example: one active and one inactive instead of returning random one.
I didn't have any other idea how to "push" the active on top, because if I wanted to use `.find` by active, it wouldn't find inactive ones if there were the only one and the validation wouldn't appear.
As I wrote in one of the comments - I think the better idea here would be to use `.find` and NEVER find inactive ones, so we don't need that validation error, but just don't assign the product source if it's inactive, but it will need to be discussed with Katarzyna and Kelsey.

As for second part of question - with negation, it sorts in order: true, false - as for active = null I don't think it would ever happen since a product source is `true` by default: `Boolean active = Boolean.TRUE`. But answering - `null` is falsy, so anyway it would return the active one on the top instead of null.